src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/config/AsyncStoragePoolConfig.java

```diff
@@ -0,0 +1,40 @@
+package com.amazon.opendistroforelasticsearch.security.auditlog.config;
+
+import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
+import org.elasticsearch.common.settings.Settings;
+
+public class AsyncStoragePoolConfig {
+    private static final int DEFAULT_THREAD_POOL_SIZE = 10;
+    private static final int DEFAULT_THREAD_POOL_MAX_QUEUE_LEN = 100 * 1000;
+
+    private final int threadPoolSize;
+    private final int threadPoolMaxQueueLen;
+
+    public AsyncStoragePoolConfig(int threadPoolSize, int threadPoolMaxQueueLen) {
+        if (threadPoolSize <= 0) {
```

Comment:
I believe it is a very common practice? 
Creating a thread pool with negative value will complain and we can go two ways from there:
- Complain and crash the process
- Use default values. 

I will log a warning for this case.

Reply:
My take is to complain (log error) and crash. IMO, silently switching to a default value will cause confusion. Logging a warning may go unnoticed.

Reply:
I m slightly not sure, we should throw an exception and make it crash.
- Reading from elasticsearch settings asks for a default value if not configured.
- Users who don't care to make a specific config setting, this will crash for them too instead of relying on defaults.
- This would make thread pool setting mandatory (may need to consult product and update docs).

Reply:
If one of the settings is not present it needs to be replaced with a default value, there should not be any crash. In case an invalid value is passed to the constructor, it should be OK to throw an exception and fail to construct `AsyncStoragePoolConfig` object.

Reply:
Alright